src/serialization.c

```diff
@@ -279,4 +331,93 @@ RRStatus RaftRedisCommandArrayDeserialize(RaftRedisCommandArray *target, const v
     return RR_OK;
 }
 
+RRStatus RaftRedisDeserializeImport(ImportKeys * target, const void *buf, size_t buf_size)
+{
+    const char *p = buf;
+    int n;
+
+    FreeImportKeys(target);
+
+    /* Read term */
+    size_t term;
+    if ((n = decodeInteger(p, buf_size, '*', &term)) < 0 || !term) {
+        return RR_ERROR;
+    }
+    p += n; buf_size -= n;
+    target->term = term;
+
+    /* Read magic */
+    size_t magic;
+    if ((n = decodeInteger(p, buf_size, '*', &magic)) < 0) {
+        return RR_ERROR;
+    }
+    p += n; buf_size -= n;
+    target->magic = (int) magic;
+
+    /* Read magic */
+    size_t num_keys;
+    if ((n = decodeInteger(p, buf_size, '*', &num_keys)) < 0 || !num_keys) {
+        return RR_ERROR;
+    }
+    p += n; buf_size -= n;
+    target->num_keys = num_keys;
+
+    target->key_names = RedisModule_Calloc(num_keys, sizeof(RedisModuleString*));
+    target->key_serialized = RedisModule_Calloc(num_keys, sizeof(RedisModuleString*));
+
+    for (size_t i = 0; i < num_keys; i++) {
+        if ((n = decodeString(p, buf_size, &target->key_names[i])) < 0 || !target->key_names[i]) {
```

Comment:
Just curious, does this ever execute second part of the if check: `!target->key_names[i]` ? If `decodeString()` returns negative, it won't be executed. If `decodeString()` is positive, that part is never null? Just consider if it makes sense.

Reply:
looking at it again.  conceptually, one could pass in a string length of 0, which would then return an empty string which would e an error.  so the check is probably useful.